modules/core/src/core/lib/layer.js

```diff
@@ -362,6 +362,28 @@ export default class Layer {
     }
   }
 
+  // this method overwrittes the current buffer and is used for multi-picking
+  updateInstancePickingColors(excludeColors) {
+    const excludeIds = excludeColors.map(this.decodePickingColor);
+
+    const {instancePickingColors} = this.getAttributeManager().attributes;
+    const {state: attribute, allocedInstances: numInstances} = instancePickingColors;
+
+    const {value, size} = attribute;
+    // add 1 to index to seperate from no selection
+    for (let i = 0; i < numInstances; i++) {
+      const pickingColor = excludeIds.includes(i)
+        ? this.nullPickingColor()
+        : this.encodePickingColor(i);
+      value[i * size + 0] = pickingColor[0];
+      value[i * size + 1] = pickingColor[1];
+      value[i * size + 2] = pickingColor[2];
+    }
+
+    // TODO: Optimize this to use sub-buffer update!
```

Comment:
Another TODO: Alternatively, we could use transform feedback to clear the specified color in the buffer.